Update Home Page and Dashboards app documentation - OSD PR4 (#12431)
opensearch-project/documentation-website: _dashboards/dashboard/dash-tutorial.md

---
layout: default
title: Dashboard tutorial
parent: Creating dashboards
nav_order: 50
has_children: false
---

# Tutorial: creating a dashboard

You can use the **Dashboards** application in OpenSearch Dashboards to build a page containing multiple visual representations of your data.

>This documentation uses the following terms:
>- _OpenSearch Dashboards_: The web UI for OpenSearch.
>- **Dashboards** application: The application within OpenSearch Dashboards for creating dashboards.
>- _dashboard_ (lowercase): An individual collection of data visualizations created in the **Dashboards** application.
{: .note}

Use the following tutorial to learn to create a dashboard using the **Dashboards** application and the OpenSearch sample data. The sample datasets have existing sample visualizations that you can use for the dashboard, or you can create your own visualizations. The tutorials demonstrate how to do both.

## Navigating the UI

The following components make up the **Dashboards** UI.

> ![Dashboards user interface]({{site.url}}{{site.baseurl}}/images/dashboards/dashboard-UI.png)

Reviewer: This page uses `dashboard-UI.png` while `index.md` uses the new `dashboard-UI-blank-callouts.png` (added in this PR) for the same callout list (A–F vs A–E). The two pages now show different screenshots and different callout schemes for the same UI — confusing for someone reading the tutorial right after `index.md`. Worth aligning on one image + one callout map.

- The _navigation panel_ (A) on the left contains the OpenSearch Dashboards applications.
- The _search_ bar (B) enables selection of data using a query language search.
- The _filter_ (C) provides a graphical interface for selecting data values and ranges.
- The _toolbar_ (D) contains frequently used commands and shortcuts.
- The _time filter_ (E) enables selection of data based on a time and date range.
- The _panel_ (F) displays the dashboard. You can add existing visualizations, create new ones, and save the dashboard's configuration.

The following tutorials assume you're either using your existing installation of OpenSearch Dashboards or using the [OpenSearch Playground](https://playground.opensearch.org/app/home#/). Depending on which one you use, certain capabilities may not be available. For example, sample datasets may not be included in your existing installation, and saving a dashboard isn't an option in the OpenSearch Playground.
{: .note}

## Prerequisites

Before starting this tutorial, ensure that you:

- Familiarize yourself with the **Dashboards** UI. See [Navigating the UI]({{site.url}}{{site.baseurl}}/dashboards/dashboard/#navigating-the-ui).

> - [Install OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/install-and-configure/install-dashboards).

Reviewer: Missing trailing slash.

```suggestion
- [Install OpenSearch Dashboards]({{site.url}}{{site.baseurl}}/install-and-configure/install-dashboards/).
```

   {: .note}

- Add sample data or import your own data into OpenSearch. To learn about adding sample datasets, see [Adding sample data]({{site.url}}{{site.baseurl}}/dashboards/quickstart/#adding-sample-data). To learn about importing your own data, see [Managing indexes]({{site.url}}{{site.baseurl}}/im-plugin/index/).

- Know how to use the [search bar]({{site.url}}{{site.baseurl}}/dashboards/discover/search-bar/) and the [time filter]({{site.url}}{{site.baseurl}}/dashboards/discover/time-filter/).

- Understand OpenSearch [documents]({{site.url}}{{site.baseurl}}/getting-started/intro/#document) and [indexes]({{site.url}}{{site.baseurl}}/getting-started/intro/#index).

## Creating a dashboard

To create a new dashboard, follow these steps:

1. | In the default navigation panel | In the New Home Page navigation panel |
   | :-- | :-- |
   | - Select **OpenSearch Dashboards** > **Dashboards**.<br/>- Select **Dashboards**. | Select **Dashboards**. |

1. | In the default navigation panel | In the New Home Page navigation panel |
   | :-- | :-- |

>    | - In the Dashboards panel, select **Create** >.<br/>- From the drop-down, select **Dashboard**. | From the application menu, select **Create Dashboard**. |

Reviewer: **Create >** is rendered with a stray trailing `>`. Either remove it or, since the next line says "From the drop-down, select **Dashboard**", merge into a single combined step. As written it looks like a typo.

## Adding an existing visualization

To add a saved visualization to the dashboard, follow these steps:

1. From the application panel, choose **Add an existing**.

1. In the **Add panels** dialog, type `ecommerce` in the **Search** box to filter the list of available visualizations.

1. From the **Add panels** dialog, choose **[eCommerce] Sales by Category**.

1. Choose the {::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/icons/cross-icon.png" class="inline-icon" alt="cross icon"/>{:/} (cross) icon to close the dialog.

1. Use the [time filter]({{site.url}}{{site.baseurl}}/dashboards/discover/time-filter/) to select `Last 2 years` as the time window as shown in the following image.

> <img src="{{site.url}}{{site.baseurl}}/images/dashboards/dash-tut-time-2-years.png" alt="Discover app default page" width="40%">

Reviewer: Wrong alt text — image is the time-filter selector, not the Discover landing page.

```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/dashboards/dash-tut-time-2-years.png" alt="Time filter set to Last 2 years" width="40%">
```

## Saving a dashboard

To save a new dashboard, follow these steps:

1. In the **Dashboards** toolbar, choose **Save**.

1. In the **Save dashboard** dialog, enter `[Ecommerce] tutorial dashboard` in the **Title** box.

1. Save the time filter with the dashboard by selecting **Store time with dashboard**.

1. Choose **Save**.

1. Choose **Edit** from the application menu to continue editing the dashboard.

## Creating a visualization

To create a new visualization in **Dashboards**, follow these steps:

1. From the application toolbar, choose **Create new**.

1. From the **New Visualization** window, choose **Metric**.

1. In the **New Metric/Choose a source** dialog, select the index pattern **opensearch_dashboards_sample_data_ecommerce**.

1. From the toolbar, choose **Save**.

1. In the **Save visualization** dialog, enter a title for the visualization. For this tutorial, enter `[eCommerce] Order Count`.

1. Choose **Save and return**.

   The **Dashboards** application saves the metric visualization and adds it to the dashboard as shown in the following image.

>    <img src="{{site.url}}{{site.baseurl}}/images/dashboards/dash-tut-combined.png" alt="Sales by category area visualization"  width="100%">

Reviewer: Alt text says "Sales by category area visualization" but the image shows the dashboard *combined* (sales-by-category + the new metric). Mismatch with what's described in the surrounding prose.

```suggestion
   <img src="{{site.url}}{{site.baseurl}}/images/dashboards/dash-tut-combined.png" alt="Dashboard with Sales by Category and Order Count metric panels"  width="100%">
```

## Adding subsequent panels

Add a Markdown visualization to the dashboard. Follow these steps:

1. From the dashboard toolbar, choose **Add**.

1. From the **Add panels** dialog, choose **[eCommerce] Markdown**.

1. Choose the {::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/icons/cross-icon.png" class="inline-icon" alt="cross icon"/>{:/} (cross) icon to close the dialog.

   The **Dashboards** application adds the Markdown panel to the dashboard as shown in the following image.

   <img src="{{site.url}}{{site.baseurl}}/images/dashboards/dash-tut-three-panel.png" alt="Example dashboard"  width="100%">

## Organizing a dashboard

You can organize a dashboard by resizing and rearranging panels. Move and resize the Markdown panel to serve as a title and description for the dashboard. Follow these steps:

1. Select and hold the top of the panel anywhere to the left of the {::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/icons/gear-icon.png" class="inline-icon" alt="options icon"/>{:/} (options) icon.

1. Drag the panel to the top of the application panel.

   The Sales by Category panel automatically swaps places with the Markdown panel as you move it up.

> 

Reviewer: This line breaks the numbered list — it's a free-floating paragraph between two `1.` items, so the renderer restarts numbering at L162 instead of continuing the procedure. Either indent it under the previous step or convert it into its own numbered step.

```suggestion
1. Select and hold the {::nomarkdown}<img src="{{site.url}}{{site.baseurl}}/images/icons/resize-icon.png" class="inline-icon" alt="resize icon"/>{:/} (resize) icon in the panel's lower-right corner.
```
